spec/requests/api/graphql/mutations/create_consignment_submission_spec.rb

```diff
@@ -0,0 +1,115 @@
+    <<-GRAPHQL
+    mutation {
+      createConsignmentSubmission(input: #{mutation_inputs}){
+        clientMutationId
+        consignmentSubmission {
+          id
+          title
+          category
+          state
+          minimum_price_dollars
+          currency
+        }
+      }
+    }
+    GRAPHQL
+  end
+
+  describe 'invalid requests' do
+    context 'with an unauthorized request' do
+      let(:token) { 'foo.bar.baz' }
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        create_response = body['data']['createConsignmentSubmission']
+        expect(create_response).to eq nil
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to eq "Can't access createConsignmentSubmission"
+      end
+    end
+
+    context 'with a request missing an app token' do
+      let(:token) do
+        payload = { sub: 'userid', roles: 'user' }
+        JWT.encode(payload, Convection.config.jwt_secret)
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        create_response = body['data']['createConsignmentSubmission']
+        expect(create_response).to eq nil
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to eq "Can't access createConsignmentSubmission"
+      end
+    end
+
+    context 'with a request missing an artist_id' do
+      let(:mutation_inputs) { '{ title: "soup" }' }
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to match 'is required'
+      end
+    end
+  end
+
+  describe 'valid requests' do
+    it 'creates a submission' do
+      stub_gravity_root
+      stub_gravity_user
+      stub_gravity_user_detail(email: 'michael@bluth.com')
+
+      expect {
+        post '/api/graphql', params: { query: mutation }, headers: headers
+      }.to change(Submission, :count).by(1)
+
+      expect(response.status).to eq 200
+      body = JSON.parse(response.body)
+      create_response = body['data']['createConsignmentSubmission']
+
+      submission_response = create_response['consignmentSubmission']
+      expect(submission_response['id']).not_to be_nil
+      expect(submission_response['title']).to eq 'soup'
+      expect(submission_response['category']).to eq 'JEWELRY'
+      expect(submission_response['state']).to eq 'REJECTED'
+      expect(submission_response['minimum_price_dollars']).to eq 50_000
+      expect(submission_response['currency']).to eq 'GBP'
```

Comment:
Rather than individual field testing with multiple `expect`s, is it possible to use a single expect and assert directly on the value (not instance) of a hash? Somethiing like:

```ruby
expect(response).to eq {
  title: 'soup',
  category: 'JEWELRY'
  ...
}
```

And then let the diff in the failure tell the story?

Reply:
Good call, updated! Turns out the better way to do this is:

```
expect(thingy).to include(
  key: 'value',
  another: 'value'
)
```

So that's probably better.